specs/Autofill.md

```diff
@@ -0,0 +1,130 @@
+
+# Examples
+
+## Win32 C++
+```cpp
+void SettingsComponent::TogglePasswordAutofill() {
+    wil::com_ptr<ICoreWebView2Settings> settings;
+    CHECK_FAILURE(webView->get_Settings(&settings));
+    wil::com_ptr<ICoreWebView2Settings4> settings4 = settings.try_query<ICoreWebView2Settings4>();
+    if (settings4) 
+    {
+        bool enabled;
+        CHECK_FAILURE(settings4->get_IsPasswordAutofillEnabled(&enabled));
+        CHECK_FAILURE(settings4->put_IsPasswordAutofillEnabled(!enabled));
+    }
+}
+
+void SettingsComponent::ToggleGeneralAutofill() {
+    wil::com_ptr<ICoreWebView2Settings> settings;
+    CHECK_FAILURE(webView->get_Settings(&settings));
+    wil::com_ptr<ICoreWebView2Settings4> settings4 = settings.try_query<ICoreWebView2Settings4>();
+    if (settings4) 
+    {
+        bool enabled;
+        CHECK_FAILURE(settings4->get_IsGeneralAutofillEnabled(&enabled));
+        CHECK_FAILURE(settings4->put_IsGeneralAutofillEnabled(!enabled));
+    } 
+}
+```
+
+## .NET, WinRT
+```c#
+
+// This demonstrates a scenario in which a button triggers TogglePasswordAutofill or 
+// ToggleGeneralAutofill. 
+
+private void TogglePasswordAutofill(CoreWebView2 sender, CoreWebView2NavigationStartingEventArgs e)
+{
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.IsPasswordAutofillEnabled = !settings.IsPasswordAutofillEnabled;
+}
+
+private void ToggleGeneralAutofill(CoreWebView2 sender, CoreWebView2NavigationStartingEventArgs e)
+{
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.IsGeneralAutofillEnabled = !settings.IsGeneralAutofillEnabled;
+}
+
+```
+
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+
+# API Details
+```IDL
+interface ICoreWebView2Settings4
+
+/// A continuation of the ICoreWebView2Settings interface.
+[uuid(cb56846c-4168-4d53-b04f-03b6d6796ff2), object, pointer_default(unique)]
+interface ICoreWebView2Settings4 : ICoreWebView2Settings3 {
+  /// IsPasswordAutosaveEnabled controls whether autosave for password
+  /// information is enabled. The IsPasswordAutosaveEnabled property behaves 
+  /// independently of the IsGeneralAutofillEnabled property. When IsPasswordAutosaveEnabled is
+  /// false, no new password data is saved and no Save/Update Password prompts are displayed. 
+  /// However, if there was password data already saved before disabling this setting, 
+  /// then that password information is auto-populated, suggestions are shown and clicking on 
+  /// one will populate the fields.
+  /// When IsPasswordAutosaveEnabled is true, password information is auto-populated, 
+  /// suggestions are shown and clicking on one will populate the fields, new data 
+  /// is saved, and a Save/Update Password prompt is displayed. 
+  /// The default value is `FALSE`.
+  [propget] HRESULT IsPasswordAutosaveEnabled([out, retval] BOOL* isPasswordAutosaveEnabled);
```

Comment:
@david-risney I keep forgetting to mention that the parameter name for property getters and setters is traditionally called just `value` (rather than repeating the property name). Too late to fix?

Reply:
Not too late! Does this mean updating the description to something like: 
This value controls whether autosave for password information is enabled. The value behaves independently of the IsGeneralAutofillEnabled property. When the value is...

Or do you mean the actual property in the method declaration:
[propget] HRESULT IsPasswordAutosaveEnabled([out, retval] BOOL* value);

Reply:
Just the name of the parameter. 

```cpp
[propget] HRESULT IsPasswordAutosaveEnabled([out, retval] BOOL* value);
```

Documentation remains unchanged.

Reply:
@david-risney are we going to go ahead and make this change for all property getters and setters in the future, or do we want to update all old property getters/setters as well?

Reply:
It's incompatible to change parameter names (though super corner in this case). I wouldn't change the existing, just adopt the pattern going forward.

Reply:
@MikeHillberg But it's okay to change the names here because these aren't the Windows Runtime IDLs. These are the raw C++ IDLs, and raw C++ doesn't care about the parameter names.